File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int n=nums.size();
       int currs=0;
       int maxs=INT_MIN;
       for(int i=0;i<n;i++){
         currs +=nums[i];

         if(maxs<currs){
            maxs=currs;
         }
         if(currs<0){
            currs=0;
         }
       }
       return maxs;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int j=0;
        int curr_sum=0;
        int max_sum=nums[0];

       while(j<nums.length){
        curr_sum +=nums[j];
        max_sum= Math.max(max_sum,curr_sum);
        if (curr_sum < 0) {
                curr_sum = 0;
            }
        j++;
       }
       return max_sum;
    }
}